python/ray/dag/compiled_dag_node.py

```diff
@@ -1250,9 +1257,12 @@ def _execute_until(
                     "on previous CompiledDAGRefs to free them up from buffer."
                 )
             self._max_execution_index += 1
+            start_time = time.monotonic()
             self._result_buffer[
                 self._max_execution_index
-            ] = self._dag_output_fetcher.read()
+            ] = self._dag_output_fetcher.read(timeout)
+            if timeout is not None:
+                timeout -= time.monotonic() - start_time
```

Comment:
I think this also needs to be `max(timeout, 0)`.

If it's not too difficult, can you also see if we can add a test for this case?

Reply:
Added test_buffered_get_timeout.